### PR TITLE
doc: release: 4.0: add notes on ZMS

### PR DESCRIPTION
**Pull request by rghaddab**
Add notes about the new storage system ZMS

### CODE REVIEW

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -598,6 +598,12 @@ Libraries / Subsystems
     from version 2.8.1, the last module update, up to and including
     the released version 2.9.3.
 
+  * ZMS: Introduction of a new storage system that is designed to work with all types of
+    non-volatile storage technologies. It supports classical on-chip NOR flash as well as
+    new technologies like RRAM and MRAM that do not require a separate erase operation at all.
+    More details can be found at:
+    https://docs.zephyrproject.org/latest/services/storage/zms/zms.html
+
```
[the remark this answers is not in the document]
Comment (rghaddab) [reply doc/releases/release-notes-4.0.rst]: @kartben I removed the permanent link and changed it with a reference to the zms_api

**Review by de-nordic — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Review by mmahadevan108 — DISMISSED** [review]

**Review by dkalowsk — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@rghaddab, please help address the merge conflict.

**Comment by rghaddab** [comment]
> @rghaddab, please help address the merge conflict.

@mmahadevan108 Done

**Review by dkalowsk — APPROVED** [review]

**Review by kartben — APPROVED** [review]
This really should be the first bullet point in this now long list, to get the attention it deserves, even if one coud argue LittleFS comes first in alphabetical order :)
We can fix this up later closer to release date when making a final sanitary pass on the document though
Thanks!

**Pull request merged**